packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3167 @@
+///   `MenuHandle`.
+/// * [MenuBar], a widget that manages its own `MenuHandle` internally.
+/// * [MenuButton], a widget that has a button that manages a submenu.
+/// * [MenuItemButton], a widget that draws a menu button with optional shortcut
+///   labels.
+class MenuHandle extends _MenuHandleBase {
+  /// Private constructor because menu entries can only be created by
+  /// [createMaterialMenu].
+  MenuHandle._({
+    _MenuHandleBase? parent,
+    required List<Widget> menuChildren,
+    FocusNode? buttonFocusNode,
+    VoidCallback? onOpen,
+    VoidCallback? onClose,
+    Offset alignmentOffset = Offset.zero,
+    Offset? globalMenuPosition,
+    ButtonStyle? buttonStyle,
+    MenuStyle? menuStyle,
+    Clip menuClipBehavior = Clip.none,
+    bool ownsParent = false,
+  })  : _parent = parent,
+        _menuChildren = menuChildren,
+        _buttonFocusNode = buttonFocusNode,
+        _onOpen = onOpen,
+        _onClose = onClose,
+        _alignmentOffset = alignmentOffset,
+        _globalMenuPosition = globalMenuPosition,
+        _buttonStyle = buttonStyle,
+        _menuStyle = menuStyle,
+        _menuClipBehavior = menuClipBehavior,
+        _ownsParent = ownsParent,
+        _menuScopeNode = FocusScopeNode() {
+    assert(() {
+      _menuScopeNode.debugLabel = 'Menu Scope';
+      return true;
+    }());
+    _parent?._addChild(this);
+  }
+
+  @override
+  final _MenuHandleBase? _parent;
+
+  @override
+  bool get isOpen => _overlayEntry != null;
+
+  @override
+  Axis get _orientation => Axis.vertical;
+
+  final List<Widget> _menuChildren;
+  final FocusScopeNode _menuScopeNode;
+  final MenuStyle? _menuStyle;
+  final ButtonStyle? _buttonStyle;
+  final VoidCallback? _onOpen;
+  final VoidCallback? _onClose;
+  final FocusNode? _buttonFocusNode;
+  final Clip _menuClipBehavior;
+  final Offset _alignmentOffset;
+  final bool _ownsParent;
+  Rect? _buttonRect;
+  Offset? _globalMenuPosition;
+
+  @protected
+  OverlayEntry? _overlayEntry;
+
+  FocusNode? get _firstItemFocusNode {
+    assert(ChangeNotifier.debugAssertNotDisposed(this));
+    if (_menuScopeNode.context == null) {
+      return null;
+    }
+    final FocusTraversalPolicy policy =
+        FocusTraversalGroup.maybeOf(_menuScopeNode.context!) ?? ReadingOrderTraversalPolicy();
+    return policy.findFirstFocus(_menuScopeNode, ignoreCurrentFocus: true);
+  }
+
+  /// Open the menu, optionally at a global position.
+  ///
+  /// Call this when the menu should be shown to the user.
+  ///
+  /// The optional `position` argument will update the global coordinate where
+  /// the menu should be opened, taking into account the [_alignmentOffset].
+  ///
+  /// If `position` is not given, and no `globalMenuPosition` was given to
+  /// [createMaterialMenu], the menu appears at the location that is
+  /// [_alignmentOffset] away from the alignment origin specified by
+  /// [MenuStyle.alignment] for the menu.
```

Comment:
I thought the LayerLink or MenuAnchor from context is used for positioning if no explicit position is given?

Reply:
Yes, that's correct.  I clarified the docs.